craft_providers/multipass/multipass.py

```diff
@@ -342,33 +342,34 @@ def transfer_source_io(
         :raises MultipassError: On error.
         """
         command = [str(self.multipass_path), "transfer", "-", destination]
-        proc = subprocess.Popen(command, stdin=subprocess.PIPE, stderr=subprocess.PIPE)
-
-        # Should never happen, but mypy/pyright makes noise.
-        assert proc.stdin is not None
-        assert proc.stderr is not None
-
-        while True:
-            data = source.read(chunk_size)
-            if not data:
-                break
-
-            proc.stdin.write(data)
-
-        # Wait until process is complete.
-        try:
-            _, stderr = proc.communicate(timeout=30)
-        except subprocess.TimeoutExpired:
-            proc.kill()
-            _, stderr = proc.communicate()
-
-        if proc.returncode != 0:
-            raise MultipassError(
-                brief=f"Failed to transfer file to destination {destination!r}.",
-                details=errors.details_from_command_error(
-                    cmd=command, stderr=stderr, returncode=proc.returncode
-                ),
-            )
+        with subprocess.Popen(
+            command, stdin=subprocess.PIPE, stderr=subprocess.PIPE
+        ) as proc:
+            # Should never happen, but mypy/pyright makes noise.
+            assert proc.stdin is not None
+            assert proc.stderr is not None
+
+            while True:
+                data = source.read(chunk_size)
+                if not data:
+                    break
+
+                proc.stdin.write(data)
+
+            # Wait until process is complete.
+            try:
+                _, stderr = proc.communicate(timeout=30)
+            except subprocess.TimeoutExpired:
+                proc.kill()
+                _, stderr = proc.communicate()
```

Comment:
Do we still need this second `communicate`, taking in consideration that all `fd`s will be closed by the context manager?

Reply:
Done.